src/components/CompareResults/loader.ts

```diff
@@ -216,13 +214,14 @@ export async function loader({ request }: { request: Request }) {
     ),
   );
 
-  const [baseRevInfo, ...newRevsInfo] = await Promise.all([
+  const [results, baseRevInfo, ...newRevsInfo] = await Promise.all([
+    resultsPromise,
     baseRevInfoPromise,
     ...newRevsInfoPromises,
   ]);
 
-  return defer({
-    results: resultsPromise,
+  return {
+    results,
```

Comment:
So I removed the `promise` here because I removed the `defer` mechanism and reverted to the previous return object - where we resolved the results in the `Promise.all`. But I can see this is why I didn't see the `Suspense ` spinner with the `Await `component regarding the results table.. Therefore, I'll put it back. I want you to understand my logic here and undersatnd I'm not this doing this willy-nilly lol.

Reply:
To make my remark clearer: by "awaiting" the result promise, there's no way the suspense fallback would appear: indeed the first render wouldn't even happen before the loader ends, but we need this first render for the suspense fallback to appear.

In short, here is the sequence of things:

1. the user presses "compare" (or change the value in the "framework" dropdown)
2. react router triggers a rerender with the same data (I believe it's a mistake but this is how it is)
3. react router runs the loader, and waits for its promise to resolve if it's a promise. (reminder that async functions are functions that return a promise)
4. if "defer" isn't used, react router triggers a new react render with the new data returned by the loader. That data can be anything, including a promise. Suspense + Await allows to wait for this promise and show a fallback before that. With the weird behavior I described in https://github.com/mozilla/perfcompare/pull/702#discussion_r1705274740.
5. If "defer" is used, react router waits for the promise(s) passed to defer to be resolved before triggering a new render. Suspense + Await is used only for the first render, not rerenders.

Hope this makes things clearer. I'm still learning how this works too, I find react-router made some weird choices.